src/common/lib/transport/connectionmanager.ts

```diff
@@ -114,22 +109,11 @@ export class TransportParams {
         break;
       case 'resume':
         params.resume = this.connectionKey as string;
-        if (this.timeSerial !== undefined) {
-          params.timeSerial = this.timeSerial as string;
-        } else if (this.connectionSerial !== undefined) {
-          params.connectionSerial = this.connectionSerial;
-        }
         break;
       case 'recover': {
-        const match = (options.recover as string).split(':');
-        if (match) {
-          params.recover = match[0];
-          const recoverSerial = match[1];
-          if (isNaN(Number(recoverSerial))) {
-            params.timeSerial = recoverSerial;
-          } else {
-            params.connectionSerial = recoverSerial;
-          }
+        const recoveryContext = RecoveryContext.decode(options.recover);
+        if (recoveryContext) {
+          params.recover = recoveryContext.connectionKey;
```

Comment:
Why are we assigning `connectionKey` to `recover`

Reply:
It seems we are using `recover` as a string at several places.

Reply:
Realtime expects the params `?<mode>=<connectionKey>` - this is the same behaviour as before

Yeah it decodes it in a few places to match the existing behaviour - can look at refactoring later

Reply:
Yeah, I mean I am not getting why `connectionKey` is being assigned to `recover`

Reply:
Since realtime expects `?recover=<connectionKey>` - where `recover` is the mode

This is the same behaviour as before

Reply:
(I was about to comment linking the relevant spec item that requires this, but I... actually can't find it? Possibly this was never properly specified. I'll add something to my docs pr...)

Reply:
That would be great!

Reply:
made the spec a lot more explicit: https://github.com/ably/docs/commit/2a932d4c9e19a9855060994bc882cc27cf23b9a7

Reply:
Thanks!!